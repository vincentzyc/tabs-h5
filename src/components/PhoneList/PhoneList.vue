<template>
  <div class="phone-list" v-if="phoneList.length > 0">
    <ul class="flex flex-wrap space-between">
      <li class="phone-item" v-for="phone in phoneList">
        <div class="flex flex-center flex-column phone-number">
          <p class="num-char">
            <strong :style="getHighlight(num.highlight)" v-for="num in phone.item">{{ num.numChar }}</strong>
          </p>
          <div class="flex align-middle space-around phone-info">
            <del class="phone-price">{{ getShowMoney(phone.showMoney) }}</del>
            <p class="phone-text"> {{ phone.showMsg }}</p>
          </div>
        </div>
      </li>
    </ul>
  </div>
  <div class="empty-tip" v-else>
    哎呦，没有适合您的号码！
    <p><small style="color: #ccc">建议缩短号码字数，例如：88</small></p>
  </div>
</template>

<script setup lang="ts">
// import phoneList from "./num";

type TypePhoneList = {
  productCode: string;
  num: string;
  showMsg: string;
  showMoney: string;
  item: {
    numChar: string;
    highlight: string;
  }[];
}[];

let phoneList = $ref<TypePhoneList>([]);

function getShowMoney(txt: string) {
  if (typeof txt !== "string" || txt === "") return "";
  if (txt.includes("元")) return txt;
  return txt + "元";
}
function getHighlight(highlight: string) {
  if (highlight === "0") return {};
  if (highlight === "1") return { color: "#fd3134" };
}
</script>
