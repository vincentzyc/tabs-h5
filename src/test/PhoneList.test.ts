import { mount } from '@vue/test-utils'
import PhoneList from '@/components/PhoneList/PhoneList.vue'

test('mount PhoneList component', async () => {
  expect(PhoneList).toBeTruthy()

  const wrapper = mount(PhoneList, {
    props: {
      list: [
        { "productCode": "", "num": "17688805942", "showMsg": "免费领取", "showMoney": "12528", "item": [{ "numChar": "1", "highlight": "0" }, { "numChar": "7", "highlight": "0" }, { "numChar": "6", "highlight": "0" }, { "numChar": "8", "highlight": "1" }, { "numChar": "8", "highlight": "1" }, { "numChar": "8", "highlight": "1" }, { "numChar": "0", "highlight": "0" }, { "numChar": "5", "highlight": "0" }, { "numChar": "9", "highlight": "0" }, { "numChar": "4", "highlight": "0" }, { "numChar": "2", "highlight": "0" }] },
        { "productCode": "", "num": "17688804349", "showMsg": "免费领取", "showMoney": "12528", "item": [{ "numChar": "1", "highlight": "0" }, { "numChar": "7", "highlight": "0" }, { "numChar": "6", "highlight": "0" }, { "numChar": "8", "highlight": "1" }, { "numChar": "8", "highlight": "1" }, { "numChar": "8", "highlight": "1" }, { "numChar": "0", "highlight": "0" }, { "numChar": "4", "highlight": "0" }, { "numChar": "3", "highlight": "0" }, { "numChar": "4", "highlight": "0" }, { "numChar": "9", "highlight": "0" }] },
        { "productCode": "", "num": "13211131147", "showMsg": "免费领取", "showMoney": "11488", "item": [{ "numChar": "1", "highlight": "1" }, { "numChar": "3", "highlight": "1" }, { "numChar": "2", "highlight": "1" }, { "numChar": "1", "highlight": "1" }, { "numChar": "1", "highlight": "1" }, { "numChar": "1", "highlight": "1" }, { "numChar": "3", "highlight": "1" }, { "numChar": "1", "highlight": "1" }, { "numChar": "1", "highlight": "1" }, { "numChar": "4", "highlight": "0" }, { "numChar": "7", "highlight": "0" }] },
        { "productCode": "", "num": "13249993378", "showMsg": "免费领取", "showMoney": "10678", "item": [{ "numChar": "1", "highlight": "0" }, { "numChar": "3", "highlight": "0" }, { "numChar": "2", "highlight": "0" }, { "numChar": "4", "highlight": "0" }, { "numChar": "9", "highlight": "1" }, { "numChar": "9", "highlight": "1" }, { "numChar": "9", "highlight": "1" }, { "numChar": "3", "highlight": "1" }, { "numChar": "3", "highlight": "1" }, { "numChar": "7", "highlight": "0" }, { "numChar": "8", "highlight": "1" }] },
        { "productCode": "", "num": "17688222405", "showMsg": "免费领取", "showMoney": "10648", "item": [{ "numChar": "1", "highlight": "0" }, { "numChar": "7", "highlight": "0" }, { "numChar": "6", "highlight": "0" }, { "numChar": "8", "highlight": "1" }, { "numChar": "8", "highlight": "1" }, { "numChar": "2", "highlight": "1" }, { "numChar": "2", "highlight": "1" }, { "numChar": "2", "highlight": "1" }, { "numChar": "4", "highlight": "0" }, { "numChar": "0", "highlight": "0" }, { "numChar": "5", "highlight": "0" }] },
        { "productCode": "", "num": "13189666446", "showMsg": "免费领取", "showMoney": "9908", "item": [{ "numChar": "1", "highlight": "0" }, { "numChar": "3", "highlight": "0" }, { "numChar": "1", "highlight": "0" }, { "numChar": "8", "highlight": "0" }, { "numChar": "9", "highlight": "0" }, { "numChar": "6", "highlight": "1" }, { "numChar": "6", "highlight": "1" }, { "numChar": "6", "highlight": "1" }, { "numChar": "4", "highlight": "1" }, { "numChar": "4", "highlight": "1" }, { "numChar": "6", "highlight": "1" }] },
        { "productCode": "", "num": "18566055596", "showMsg": "免费领取", "showMoney": "9848", "item": [{ "numChar": "1", "highlight": "0" }, { "numChar": "8", "highlight": "0" }, { "numChar": "5", "highlight": "0" }, { "numChar": "6", "highlight": "1" }, { "numChar": "6", "highlight": "1" }, { "numChar": "0", "highlight": "0" }, { "numChar": "5", "highlight": "1" }, { "numChar": "5", "highlight": "1" }, { "numChar": "5", "highlight": "1" }, { "numChar": "9", "highlight": "0" }, { "numChar": "6", "highlight": "0" }] },
        { "productCode": "", "num": "13249994474", "showMsg": "免费领取", "showMoney": "9808", "item": [{ "numChar": "1", "highlight": "0" }, { "numChar": "3", "highlight": "0" }, { "numChar": "2", "highlight": "0" }, { "numChar": "4", "highlight": "0" }, { "numChar": "9", "highlight": "1" }, { "numChar": "9", "highlight": "1" }, { "numChar": "9", "highlight": "1" }, { "numChar": "4", "highlight": "1" }, { "numChar": "4", "highlight": "1" }, { "numChar": "7", "highlight": "1" }, { "numChar": "4", "highlight": "1" }] },
        { "productCode": "", "num": "13249993385", "showMsg": "免费领取", "showMoney": "9678", "item": [{ "numChar": "1", "highlight": "0" }, { "numChar": "3", "highlight": "0" }, { "numChar": "2", "highlight": "0" }, { "numChar": "4", "highlight": "0" }, { "numChar": "9", "highlight": "1" }, { "numChar": "9", "highlight": "1" }, { "numChar": "9", "highlight": "1" }, { "numChar": "3", "highlight": "1" }, { "numChar": "3", "highlight": "1" }, { "numChar": "8", "highlight": "0" }, { "numChar": "5", "highlight": "0" }] },
        { "productCode": "", "num": "13192666449", "showMsg": "免费领取", "showMoney": "9608", "item": [{ "numChar": "1", "highlight": "0" }, { "numChar": "3", "highlight": "0" }, { "numChar": "1", "highlight": "0" }, { "numChar": "9", "highlight": "0" }, { "numChar": "2", "highlight": "0" }, { "numChar": "6", "highlight": "1" }, { "numChar": "6", "highlight": "1" }, { "numChar": "6", "highlight": "1" }, { "numChar": "4", "highlight": "1" }, { "numChar": "4", "highlight": "1" }, { "numChar": "9", "highlight": "0" }] },
        { "productCode": "", "num": "13249993369", "showMsg": "免费领取", "showMoney": "9558", "item": [{ "numChar": "1", "highlight": "0" }, { "numChar": "3", "highlight": "0" }, { "numChar": "2", "highlight": "0" }, { "numChar": "4", "highlight": "0" }, { "numChar": "9", "highlight": "1" }, { "numChar": "9", "highlight": "1" }, { "numChar": "9", "highlight": "1" }, { "numChar": "3", "highlight": "1" }, { "numChar": "3", "highlight": "1" }, { "numChar": "6", "highlight": "0" }, { "numChar": "9", "highlight": "0" }] }
      ],
    },
  })
  expect(wrapper.html()).toMatchSnapshot()
  
  const listLength = wrapper.props().list.length

  expect(wrapper.findAll('.phone-item')).toHaveLength(listLength)

  await wrapper.get('.phone-item').trigger('click')
  
  const emitEvent = wrapper.emitted('handleSelectedNum')

  expect(emitEvent).toHaveLength(1)
})
