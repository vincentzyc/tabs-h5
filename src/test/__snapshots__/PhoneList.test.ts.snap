// Vitest Snapshot v1

exports[`mount PhoneList component 1`] = `
"<div class=\\"phone-list\\">
  <ul class=\\"flex flex-wrap space-between\\">
    <li class=\\"phone-item\\">
      <div class=\\"flex flex-center flex-column phone-number\\">
        <p class=\\"num-char\\"><strong>1</strong><strong>7</strong><strong>6</strong><strong style=\\"color: rgb(253, 49, 52);\\">8</strong><strong style=\\"color: rgb(253, 49, 52);\\">8</strong><strong style=\\"color: rgb(253, 49, 52);\\">8</strong><strong>0</strong><strong>5</strong><strong>9</strong><strong>4</strong><strong>2</strong></p>
        <div class=\\"flex align-middle space-around phone-info\\"><del class=\\"phone-price\\">12528元</del>
          <p class=\\"phone-text\\">免费领取</p>
        </div>
      </div>
    </li>
    <li class=\\"phone-item\\">
      <div class=\\"flex flex-center flex-column phone-number\\">
        <p class=\\"num-char\\"><strong>1</strong><strong>7</strong><strong>6</strong><strong style=\\"color: rgb(253, 49, 52);\\">8</strong><strong style=\\"color: rgb(253, 49, 52);\\">8</strong><strong style=\\"color: rgb(253, 49, 52);\\">8</strong><strong>0</strong><strong>4</strong><strong>3</strong><strong>4</strong><strong>9</strong></p>
        <div class=\\"flex align-middle space-around phone-info\\"><del class=\\"phone-price\\">12528元</del>
          <p class=\\"phone-text\\">免费领取</p>
        </div>
      </div>
    </li>
    <li class=\\"phone-item\\">
      <div class=\\"flex flex-center flex-column phone-number\\">
        <p class=\\"num-char\\"><strong style=\\"color: rgb(253, 49, 52);\\">1</strong><strong style=\\"color: rgb(253, 49, 52);\\">3</strong><strong style=\\"color: rgb(253, 49, 52);\\">2</strong><strong style=\\"color: rgb(253, 49, 52);\\">1</strong><strong style=\\"color: rgb(253, 49, 52);\\">1</strong><strong style=\\"color: rgb(253, 49, 52);\\">1</strong><strong style=\\"color: rgb(253, 49, 52);\\">3</strong><strong style=\\"color: rgb(253, 49, 52);\\">1</strong><strong style=\\"color: rgb(253, 49, 52);\\">1</strong><strong>4</strong><strong>7</strong></p>
        <div class=\\"flex align-middle space-around phone-info\\"><del class=\\"phone-price\\">11488元</del>
          <p class=\\"phone-text\\">免费领取</p>
        </div>
      </div>
    </li>
    <li class=\\"phone-item\\">
      <div class=\\"flex flex-center flex-column phone-number\\">
        <p class=\\"num-char\\"><strong>1</strong><strong>3</strong><strong>2</strong><strong>4</strong><strong style=\\"color: rgb(253, 49, 52);\\">9</strong><strong style=\\"color: rgb(253, 49, 52);\\">9</strong><strong style=\\"color: rgb(253, 49, 52);\\">9</strong><strong style=\\"color: rgb(253, 49, 52);\\">3</strong><strong style=\\"color: rgb(253, 49, 52);\\">3</strong><strong>7</strong><strong style=\\"color: rgb(253, 49, 52);\\">8</strong></p>
        <div class=\\"flex align-middle space-around phone-info\\"><del class=\\"phone-price\\">10678元</del>
          <p class=\\"phone-text\\">免费领取</p>
        </div>
      </div>
    </li>
    <li class=\\"phone-item\\">
      <div class=\\"flex flex-center flex-column phone-number\\">
        <p class=\\"num-char\\"><strong>1</strong><strong>7</strong><strong>6</strong><strong style=\\"color: rgb(253, 49, 52);\\">8</strong><strong style=\\"color: rgb(253, 49, 52);\\">8</strong><strong style=\\"color: rgb(253, 49, 52);\\">2</strong><strong style=\\"color: rgb(253, 49, 52);\\">2</strong><strong style=\\"color: rgb(253, 49, 52);\\">2</strong><strong>4</strong><strong>0</strong><strong>5</strong></p>
        <div class=\\"flex align-middle space-around phone-info\\"><del class=\\"phone-price\\">10648元</del>
          <p class=\\"phone-text\\">免费领取</p>
        </div>
      </div>
    </li>
    <li class=\\"phone-item\\">
      <div class=\\"flex flex-center flex-column phone-number\\">
        <p class=\\"num-char\\"><strong>1</strong><strong>3</strong><strong>1</strong><strong>8</strong><strong>9</strong><strong style=\\"color: rgb(253, 49, 52);\\">6</strong><strong style=\\"color: rgb(253, 49, 52);\\">6</strong><strong style=\\"color: rgb(253, 49, 52);\\">6</strong><strong style=\\"color: rgb(253, 49, 52);\\">4</strong><strong style=\\"color: rgb(253, 49, 52);\\">4</strong><strong style=\\"color: rgb(253, 49, 52);\\">6</strong></p>
        <div class=\\"flex align-middle space-around phone-info\\"><del class=\\"phone-price\\">9908元</del>
          <p class=\\"phone-text\\">免费领取</p>
        </div>
      </div>
    </li>
    <li class=\\"phone-item\\">
      <div class=\\"flex flex-center flex-column phone-number\\">
        <p class=\\"num-char\\"><strong>1</strong><strong>8</strong><strong>5</strong><strong style=\\"color: rgb(253, 49, 52);\\">6</strong><strong style=\\"color: rgb(253, 49, 52);\\">6</strong><strong>0</strong><strong style=\\"color: rgb(253, 49, 52);\\">5</strong><strong style=\\"color: rgb(253, 49, 52);\\">5</strong><strong style=\\"color: rgb(253, 49, 52);\\">5</strong><strong>9</strong><strong>6</strong></p>
        <div class=\\"flex align-middle space-around phone-info\\"><del class=\\"phone-price\\">9848元</del>
          <p class=\\"phone-text\\">免费领取</p>
        </div>
      </div>
    </li>
    <li class=\\"phone-item\\">
      <div class=\\"flex flex-center flex-column phone-number\\">
        <p class=\\"num-char\\"><strong>1</strong><strong>3</strong><strong>2</strong><strong>4</strong><strong style=\\"color: rgb(253, 49, 52);\\">9</strong><strong style=\\"color: rgb(253, 49, 52);\\">9</strong><strong style=\\"color: rgb(253, 49, 52);\\">9</strong><strong style=\\"color: rgb(253, 49, 52);\\">4</strong><strong style=\\"color: rgb(253, 49, 52);\\">4</strong><strong style=\\"color: rgb(253, 49, 52);\\">7</strong><strong style=\\"color: rgb(253, 49, 52);\\">4</strong></p>
        <div class=\\"flex align-middle space-around phone-info\\"><del class=\\"phone-price\\">9808元</del>
          <p class=\\"phone-text\\">免费领取</p>
        </div>
      </div>
    </li>
    <li class=\\"phone-item\\">
      <div class=\\"flex flex-center flex-column phone-number\\">
        <p class=\\"num-char\\"><strong>1</strong><strong>3</strong><strong>2</strong><strong>4</strong><strong style=\\"color: rgb(253, 49, 52);\\">9</strong><strong style=\\"color: rgb(253, 49, 52);\\">9</strong><strong style=\\"color: rgb(253, 49, 52);\\">9</strong><strong style=\\"color: rgb(253, 49, 52);\\">3</strong><strong style=\\"color: rgb(253, 49, 52);\\">3</strong><strong>8</strong><strong>5</strong></p>
        <div class=\\"flex align-middle space-around phone-info\\"><del class=\\"phone-price\\">9678元</del>
          <p class=\\"phone-text\\">免费领取</p>
        </div>
      </div>
    </li>
    <li class=\\"phone-item\\">
      <div class=\\"flex flex-center flex-column phone-number\\">
        <p class=\\"num-char\\"><strong>1</strong><strong>3</strong><strong>1</strong><strong>9</strong><strong>2</strong><strong style=\\"color: rgb(253, 49, 52);\\">6</strong><strong style=\\"color: rgb(253, 49, 52);\\">6</strong><strong style=\\"color: rgb(253, 49, 52);\\">6</strong><strong style=\\"color: rgb(253, 49, 52);\\">4</strong><strong style=\\"color: rgb(253, 49, 52);\\">4</strong><strong>9</strong></p>
        <div class=\\"flex align-middle space-around phone-info\\"><del class=\\"phone-price\\">9608元</del>
          <p class=\\"phone-text\\">免费领取</p>
        </div>
      </div>
    </li>
    <li class=\\"phone-item\\">
      <div class=\\"flex flex-center flex-column phone-number\\">
        <p class=\\"num-char\\"><strong>1</strong><strong>3</strong><strong>2</strong><strong>4</strong><strong style=\\"color: rgb(253, 49, 52);\\">9</strong><strong style=\\"color: rgb(253, 49, 52);\\">9</strong><strong style=\\"color: rgb(253, 49, 52);\\">9</strong><strong style=\\"color: rgb(253, 49, 52);\\">3</strong><strong style=\\"color: rgb(253, 49, 52);\\">3</strong><strong>6</strong><strong>9</strong></p>
        <div class=\\"flex align-middle space-around phone-info\\"><del class=\\"phone-price\\">9558元</del>
          <p class=\\"phone-text\\">免费领取</p>
        </div>
      </div>
    </li>
  </ul>
</div>"
`;
